### PR TITLE
[cker] Rename derivative to grad or incoming

### PR DESCRIPTION
**Pull request by ragmani**
This commit renames using the term "derivative" to the term "grad" or "incoming".

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: compute/cker/include/cker/train/operation/MaxPool.h [hunk 1]
```diff
@@ -126,22 +126,23 @@ inline void MaxPool2D(const PoolParams &params, const Shape &input_shape, const
   out_mat.cwiseMin(params.float_activation_min).cwiseMax(params.float_activation_max);
 }
 
-inline void MaxPool2DGrad(const Shape &deriv_output_shape, const float *deriv_output_data,
-                          const int *arg_max_index, const Shape &deriv_input_shape,
-                          float *deriv_input_data)
+inline void MaxPool2DGrad(const Shape &incoming_output_shape, const float *incoming_output_data,
+                          const int *arg_max_index, const Shape &back_prop_input_shape,
+                          float *back_prop_input_data)
```
Comment (zetwhite) [inline compute/cker/include/cker/train/operation/MaxPool.h:131]: I re-read this part. 
I feel that the `output` and `input` is a bit hard to catch. 

  * `output-derivation` means derivation which is corresponded to my(layer's) output-tensor. 
  * `input-derivation` means derivation which is corresponded to my(layer's) input-tensor.

For `Cker` level (Not in `Layer` Level), under naming looks better.  
```suggestion
inline void MaxPool2DGrad(const Shape &incoming_shape, const float *incoming_data,
                          const int *arg_max_index, const Shape &grad_shape,
                          float *grad_data)
```

Comment (zetwhite) [reply compute/cker/include/cker/train/operation/MaxPool.h:131]: > In Cker level (Not in Layer Level), For clarify, under naming looks better. 

@ragmani Could you share your opinion about this?
If you agree with me, How about close this PR and Let me re-name with just `incoming` and `grad`.

Comment (ragmani) [reply compute/cker/include/cker/train/operation/MaxPool.h:131]: Your suggestion looks better. Thanks, I will change it.

Comment (ragmani) [reply compute/cker/include/cker/train/operation/MaxPool.h]: I renamed it to incoming or grad.

**Review by zetwhite — APPROVED** [review]
LGTM :+1: 
Thank you!

**Review by glistening — APPROVED** [review]

**Pull request merged**